db/blip_handler.go

```diff
@@ -124,11 +126,37 @@ func (bh *blipHandler) refreshUser() error {
 	return nil
 }
 
+// userBlipHandler wraps another blip handler to specify a collection
+func collectionBlipHandler(next blipHandlerFunc) blipHandlerFunc {
+	return func(bh *blipHandler, bm *blip.Message) error {
+		collectionIndexStr, ok := bm.Properties[BlipCollection]
+		if !ok {
+			bh.collection = bh.db
+			return nil
+		}
+		collectionIndex, err := strconv.Atoi(collectionIndexStr)
+		if err != nil {
+			return base.HTTPErrorf(http.StatusBadRequest, "collection property needs to be an int, was %q", collectionIndexStr)
+		}
+
+		if len(bh.collectionMapping) > collectionIndex {
+			return base.HTTPErrorf(http.StatusBadRequest, "Passing collection requires calling GetCollections first")
+		}
```

Comment:
Should this be `if len(bh.collectionMapping) == 0` ?

If you have 1 collection, the index could be `0` and we'd still fall into this case with the current code.